Snakefile

```diff
@@ -1402,6 +1426,92 @@ rule generate_methods_differential_rnaseq:
         cp {output.methods} {wildcards.accession}-analysis-methods.tsv
         """
 
+rule deconvolution:
+    """
+    Runs three deconvolution tools (DWLS, FARDEEP, EpiDIS) for experiments selected in deconvolution.yaml.
+    """
+    conda: "envs/deconvolution.yaml"
+    log: "logs/{accession}-deconvolution.log"
+    #resources: mem_mb=get_mem_mb
+    resources: mem_mb=64000
+    threads: 16
+    input: 
+        fpkms="{accession}-fpkms.tsv.undecorated",
+	methods=get_methods_file,
+	#methods=rules.generate_methods_baseline_rnaseq.output,
+        sdrf=get_sdrf()
+    params:
+        signature_dir=config["deconv_ref"] + get_organism()
+    output:
+        proportions="{accession}-deconvolution.proportions.tsv",
+	methods="{accession}-analysis-methods.tsv",
+        results=(directory('Output/{accession}')),
+        splits=temp(directory('Tissue_splits/{accession}')),
+        scratch=temp(directory('scratch/{accession}'))
+    shell:
+        """
+        exec &> "logs/{wildcards.accession}-deconvolution.log"
+	set -b  #Notify of job termination immediately
+
+	echo "starting..."
+	# Split fpkms into organism parts
+	mkdir -p Tissue_splits/{wildcards.accession}
+	Rscript {workflow.basedir}/atlas-analysis/deconvolution/splitAndScale.R {input.fpkms} {input.sdrf} {wildcards.accession}
+
+	# list all files that FPKMs were split into
+	files=$(ls Tissue_splits/{wildcards.accession}/{wildcards.accession}*-fpkms_scaled.rds)
+
+	# iterate through tissues 
+	for file in ${{files[@]}}; do
+	    # get tissue name from filename
+	    file=$(basename "$file")
+	    tissue="${{file#{wildcards.accession}-}}"
+	    tissue="${{tissue%-fpkms_scaled.rds}}"
+	    printf "\n__________________ $tissue ____________________ \n"
+	    echo "trying to deconvolve $tissue from {wildcards.accession}"
+
+	    # search for suitable reference in deconvolution library
+	    REFERENCE_FOUND=$(Rscript {workflow.basedir}/atlas-analysis/deconvolution/findReference.R $tissue {params.signature_dir} {workflow.basedir})
+
+	    # check if a refererence for this organism part was found
+	    if [[ "$REFERENCE_FOUND" == "noref"* ]]; then
+	        echo "no reference for $tissue found"
+		sc_reference_C1="noref"
+		DECONV_STATUS="no_reference_for_deconvolution_found"
+	    else
+		# check if reference library files are okay for this tissue
+		number=$(ls {params.signature_dir}/${{REFERENCE_FOUND}}* | wc -l)
+
+		if [ "$number" != 4 ]; then
```

Comment:
why 4?

Reply:
@pmb59 Because for each UBERON organism part we have four different files (for the different tools). Added a comment